htdocs/compta/facture/class/facture-rec.class.php

```diff
@@ -1426,7 +1427,7 @@ public function createRecurringInvoices($restrictioninvoiceid = 0, $forcevalidat
 
 		$this->output = trim($this->output);
 
-		return $error ? $error : 0;
+		return count($this->errors);
```

Comment:
this->errors may not be always set, sometimes it is ->error that is set only.

If you move $error = 0; before the loop, is it ok ?